### DISCUSSION TITLE
How to use cli option value to parametrize tests?

### DISCUSSION THREAD

**Discussion by DenisKuplyakov** [pytest-dev/pytest#11147] [Q&A]
I need to read some JSON data file and use it's keys to parametrize tests. It looks like a right way is to provide path to this file through command line options. [Here](https://docs.pytest.org/en/7.3.x/example/simple.html#pass-different-values-to-a-test-function-depending-on-command-line-options) I've found a recommendation on how to deal with cli options. But the recommended way is to store cli option value as fixture, so I can't further use it to parametrize tests. 

By the moment I reverted to usage of env variables for the task. But, I am curious, is there any way to reach the goal by cli option?

```python
# conftest.py
import json

from pytest import fixture


def pytest_addoption(parser):
    parser.addoption('--data-file', type=str, default="data.json", help="data file")


# this is recommended way in pytest docs:
# https://docs.pytest.org/en/7.1.x/example/simple.html
# but there is no way to use data fixture to parametrize tests
@fixture(scope="session")
def data_file(request):
    return request.config.getoption("--data-file")


@fixture(scope="session")
def data(data_file):
    with open(data_file) as f:
        return json.load(f)

# -------------------------------------
# test.py
from pytest import mark


# FIXME: can't use data here
@mark.parametrize("key", data.keys())
def test(key):
    assert key == data[key]

# -------------------------------------
# data.json
{
  "foo": "foo",
  "bar": "bar"
}
```

**Comment by RonnyPfannschmidt** [comment]
In these cases it's best to use the pytest-configure hook to register a plugins class with temporary declared fixtures and having the data loaded there
